### PR TITLE
Merge duplicate vertices

### PR DESCRIPTION
**Pull request by lasalvavida**
Adds stage for merging duplicate vertices from the #1 roadmap.

### CODE REVIEW

**Comment by lilleyse** [comment]
For additional testing purposes, do we have any glTF models that have duplicate vertices?

File: lib/mergeDuplicateVertices.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+var writeBufferComponentType = require('./writeBufferComponentType');
+var removeUnusedVertices = require('./removeUnusedVertices');
+
+module.exports = mergeDuplicateVertices;
+
+/**
+ * Looks for duplicated vertices in a glTF hierarchy and merges them,
+ * modifying indices to account for the change if they exist.
+ *
+ * @param {Object} [gltf] A glTF hierarchy.
```
Comment (lilleyse) [inline lib/mergeDuplicateVertices.js]: `gltf` isn't optional, so remove the square braces.

**Comment by lilleyse** [comment]
The code and tests look good to me, I just want to test on a real use case before merging. I'll try to create a model for that.

**Comment by lasalvavida** [comment]
@lilleyse It looks like BrainStem has duplicate vertices, so I'm running some tests on that now.

**Comment by lilleyse** [comment]
Ok nice.

**Comment by lasalvavida** [comment]
@lilleyse Take a look at these changes when you get a chance. The [BrainStem model](https://github.com/KhronosGroup/glTF/tree/master/sampleModels/BrainStem/glTF-Embedded) should go from 15,430 KB down to 4,437 KB (with combine primitives commented out because of #85) with no detrimental effects.

**Comment by lilleyse** [comment]
Changes looks good. Also that's a great improvement! I confirmed that BrainStem works.

**Pull request merged**